optimum/exporters/onnx/config.py

```diff
@@ -94,13 +95,23 @@ def __init__(
     def inputs(self) -> Dict[str, Dict[int, str]]:
         if self.use_past_in_inputs:
             common_inputs = {"input_ids": {0: "batch_size", 1: "sequence_length"}}
+            common_inputs["attention_mask"] = {0: "batch_size", 1: "past_sequence_length + sequence_length"}
             self.add_past_key_values(common_inputs, direction="inputs")
-            common_inputs["attention_mask"] = {0: "batch_size", 1: "past_sequence_length + 1"}
         else:
             common_inputs = {
                 "input_ids": {0: "batch_size", 1: "sequence_length"},
                 "attention_mask": {0: "batch_size", 1: "sequence_length"},
             }
+
+        # Decoders based on GPT2 require a position_ids input to avoid generating wrong position_ids in the model itself:
+        # https://github.com/huggingface/transformers/blob/v4.33.1/src/transformers/models/gpt2/modeling_gpt2.py#L802
+        if (
+            not self.legacy
+            and self.task in ["text-generation", "feature-extraction"]
+            and self._config.model_type in MODEL_TYPES_REQUIRING_POSITION_IDS
+        ):
+            common_inputs["position_ids"] = {0: "batch_size", 1: "sequence_length"}
```

Comment:
the idea here is to have a single source of truth for whether a model uses/requires position ids, because otherwise it is possible to inherit from a class that inherits from `TextDecoderWithPositionIdsOnnxConfig` class, without adding the model to MODEL_TYPES_REQUIRING_POSITION_IDS, which is confusing.